### PR TITLE
Update ZIO, http4s

### PR DESCRIPTION
**Pull request by jeremyrsmith**

### CODE REVIEW

**Comment by jeremyrsmith** [comment]
Looks like we are running into this https://github.com/zio/zio/issues/2859 – we'll wait for RC18 which fixes it.

File: polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala [hunk 1]
```diff
@@ -378,8 +375,12 @@ object SocketTransport {
...
       incomingLengthBuffer.rewind()
       while(incomingLengthBuffer.hasRemaining) {
-        if(socketChannel.read(incomingLengthBuffer) == -1) {
-          return None
+        try {
+          if(socketChannel.read(incomingLengthBuffer) == -1) {
+            return None
+          }
+        } catch {
+          case err: IOException => return None
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala]: Is the error already logged somewhere else?

File: polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala [hunk 2]
```diff
@@ -151,22 +145,25 @@ object SocketTransportServer {
...
+    _        <- channel2.awaitClosed.to(closed).forkDaemon
   } yield new SocketTransportServer(server, channels, process, closed)
 }
 
 class SocketTransportClient private (channels: SocketTransport.Channels, closed: Promise[Throwable, Unit]) extends TransportClient {
 
   def logError(fn: Cause[Throwable] => ZIO[Logging, Nothing, Unit]): TaskB ~> TaskB = new ~>[TaskB, TaskB] {
-    override def apply[A](fa: TaskB[A]): TaskB[A] = fa.onError(fn)
+    override def apply[A](fa: TaskB[A]): TaskB[A] = fa.onError {
+      case cause if cause.interruptedOnly => ZIO.unit
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala]: What does this change do? If we interrupted the effect it won't log it as an error? Or something else?

File: polynote-kernel/src/main/scala/polynote/kernel/task/TaskManager.scala [hunk 3]
```diff
@@ -0,0 +1,15 @@
+package polynote.kernel.task
```
Comment (jonathanindig) [inline polynote-kernel/src/main/scala/polynote/kernel/task/TaskManager.scala]: is this file needed?

Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/task/TaskManager.scala]: Nope!

File: polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala (hunk 2 above)
Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala]: Yup!

File: polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala (hunk 1 above)
Comment (jeremyrsmith) [reply polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala]: Hmm, I guess it wouldn't be. I think this is left over from debugging, I'll revert it.

File: polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala (hunk 2 above)
Comment (jonathanindig) [reply polynote-kernel/src/main/scala/polynote/kernel/remote/transport.scala]: Nice!

**Comment by jeremyrsmith** [comment]
Closed in favor of #832
